llama_stack/providers/utils/common/data_schema_validator_mixin.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from enum import Enum
+from typing import Any, Dict, List
+
+from llama_stack.apis.common.type_system import (
+    ChatCompletionInputType,
+    CompletionInputType,
+    StringType,
+)
+
+
+class ColumnName(Enum):
+    input_query = "input_query"
+    expected_answer = "expected_answer"
+    chat_completion_input = "chat_completion_input"
+    completion_input = "completion_input"
+    generated_answer = "generated_answer"
+    context = "context"
+
+
+class DataSchemaValidatorMixin:
+    def validate_dataset_schema_for_scoring(self, dataset_schema: Dict[str, Any]):
+        self.validate_dataset_schema(
+            dataset_schema, self.get_expected_schema_for_scoring()
+        )
+
+    def validate_dataset_schema_for_eval(self, dataset_schema: Dict[str, Any]):
+        self.validate_dataset_schema(
+            dataset_schema, self.get_expected_schema_for_eval()
+        )
+
+    def validate_row_schema_for_scoring(self, row_schema: Dict[str, Any]):
+        self.validate_row_schema(row_schema, self.get_expected_schema_for_scoring())
+
+    def validate_row_schema_for_eval(self, row_schema: Dict[str, Any]):
+        self.validate_row_schema(row_schema, self.get_expected_schema_for_eval())
+
+    def get_expected_schema_for_scoring(self):
+        return [
```

Comment:
We will likely need to revisit expected data schema when evaluating for tools.

Reply:
its better for scoring/evals/post_training to all define their own schema for what they want the dataset to obey. The actual checking of the schema can be done in datasetio.

Reply:
Good idea, I'll move the schema definition out of the `DataSchemaValidatorMixin`. 

But will need some thoughts on whether we should keep `DataSchemaValidatorMixin` or expose `DataSchemaValidatorMixin` via datasetio API. Since `DataSchemaValidatorMixin` is only used internally, while datasetio API would be exposed to users.